src/common.rs

```diff
@@ -355,6 +355,130 @@ pub struct LoadAvg {
     pub fifteen: f64,
 }
 
+#[cfg(not(target_os = "windows"))]
+pub mod uid {
```

Comment:
I just thought: instead of writing this code twice, why not creating a macro taking the type (`libc::uid_t` for `not(windows)` targets) and then just call the macro twice? That would prevent code duplication and reduce its size too.

Reply:
Done, my macrofu isn't super strong so let me know if you'd like to see a different implementation